sale_commission_check_deposit/wizard/wizard_check_settle.py

```diff
@@ -0,0 +1,43 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+import json
+from datetime import date, timedelta
+
+from odoo import models
+
+
+class SaleCommissionMakeSettle(models.TransientModel):
+    _inherit = "sale.commission.make.settle"
+
+    def _get_agent_lines(self, agent, date_to_agent):
+        """
+        Exclude invoices with checks payment if at least safety days
+        haven't passed since expiration date set on check deposit.
+        """
+        agent_lines = super()._get_agent_lines(agent, date_to_agent)
+        for line in agent_lines.filtered(
+            lambda r: r.commission_id.invoice_state == "paid"
+            and r.invoice_id.invoice_payments_widget != "false"
+        ):
+            if not self.get_move_ids(line.invoice_id.invoice_payments_widget):
```

Comment:
I need to remove agent lines if it return False because if it match that conditions it means that the check is correctly deposited and safety days are passed. If not it means that I'm consider the invoice not payed yet and so I need to remove agent line.

Reply:
than use `all` in place of `any` so that will be `True` only if check is correctly deposited and all the lines passed safety days